### ISSUE TITLE
Outputs that are larger than 2 MB do not work with `share=True`

### ISSUE THREAD

**Issue by MrNeon** [gradio-app/gradio#2260]
### Describe the bug

When using the shared tunneled link I get this error when receiving outputs that are around 2MB or larger.

Error in Chrome: net::ERR_HTTP2_PROTOCOL_ERROR 200
![Screenshot_20220914_140154](https://user-images.githubusercontent.com/7870061/190165237-fb1bd7eb-5f09-44a3-8230-27e2458560b3.png)

Firefox errors too but without any code, just an uncaught exception.

 When using localhost or using ngrok to tunnel the connection the issue never happens.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```python
import numpy as np
import gradio as gr
from PIL import Image

im = Image.open("test_image_over_2MB.png")

def get_image():
    return im

with gr.Blocks() as demo:
    image_output = gr.Image()
    image_button = gr.Button("Get")
    image_button.click(get_image, inputs=None, outputs=image_output)

demo.launch(share=True, debug=True)
```

### Logs

```shell
No logs
```

### System Info

```shell
Gradio version: 3.3
Firefox 104.0
Chrome 105.0.5195.52
```

### Severity

annoying

**Comment by abidlabs** [comment]
Thanks @MrNeon for the interesting issue. This might be a limitation based on how the sharing servers have been configured. Will take a look and get back to you

**Comment by dylancl** [comment]
Is there any update on this?

**Comment by abidlabs** [comment]
I've been able to reproduce the issue, looking into a possible fix

**Comment by guumaster** [comment]
Not sure if this helps, but if you use ngrok inside Google Colab and use it to point to localhost:7860 and then access through ngrok url instead of gradio.app, it never hangs. Here is the code:

```
import os
from flask import Flask
from pyngrok import ngrok

ngrok.set_auth_token(ngrok_token) #PUT TOKEN HERE
public_url = ngrok.connect(7860, bind_tls=True).public_url

os.environ["FLASK_ENV"] = "development"
app = Flask(__name__)
app.config["BASE_URL"] = public_url
app.run()
```

Maybe it helps to see what ngrok is doing differently, unless is a server setting 🤷

**Comment by R4ZZ3** [comment]
Any estimate when this fix will be live?

**Comment by abidlabs** [comment]
Hi @R4ZZ3 we're still ironing out some issues related to the FRP branch -- but if you would like to test it out (no guarantee on its stability!) you can do it by installing `gradio` locally, checking out the branch, and then running this command: 

`cp gradio/templates/frontend/index.html gradio/templates/frontend/share.html`

before you run your Gradio app

**Comment by abidlabs** [comment]
Hi everyone! We've now officially merged in #2509, which should increase the file limits of what can be shared with public links, i.e. `share=True`. We've tested files up to 75 MB and they work successfully (although with increased latency). We'll be releasing on pypi shortly but if you'd like to try it out, you can install a beta version: `pip install gradio==3.12.0b7`

**Comment by ShadeRedeye** [comment]
i don't know if this relate to a problem,but i got these error whenever progress active,and cause output preview didn't show up,may know why like this happen?

![Screenshot_1](https://user-images.githubusercontent.com/84408428/235517651-be385c14-bcda-4424-9422-c163401671e2.png)
